daemon/daemon_main.go

```diff
@@ -1328,6 +1331,19 @@ func runDaemon() {
 		}()
 	}
 
+	// This must be before the API is served to prevent endpoints from being
+	// created before we migrate to the new ENI datapath. It also must be
+	// before the health initialization logic which might create the health
+	// endpoint, for the same reasons as being before the API is served. We
+	// want to ensure that this migration logic runs before any endpoint is
+	// being created.
+	if option.Config.IPAM == option.IPAMENI {
+		if errs := linuxrouting.MigrateENIRoutingTable(&interfaceNumberGetter{}); len(errs) > 0 {
+			log.WithField("errors", errs).
```

Comment:
Maybe we should rethink this a bit to provide better guarantees:
1. I think we want to fatal if we can't do successful migration. Datapath flow will continue and user can roll back immediately.
2. Once we have done migration but have not enabled it yet, we want to have a single atomic operation such as removing the old rules that is incredibly likely to succeed. This operation is likely reversible so we can re-add the rules and then still fatal to indicate that migration is not possible so user can rollback.

I'll add comments separately to indicate what it would take to get there.